README.md

```diff
@@ -1,2 +1,262 @@
+
+Currently, the industry standard[^2] is to use the base sZTP protocol
+ for bootstrapping devices.  Current sZTP implementations require vendor
+ specific definitions for providing the bootstrapping of a device, because
+ there is no agreed upon vendor neutral data set of “boot configuration”
+ defined. `Bootz` provides a specification that enumerates the data elements
+ which can be used in a mainly vendor agnostic way, along with the operations
+ that are required at turn-up time that become part of the “boot” process.
+ Specifically, we are suggesting that TPM-related enrollment and attestation
+ should be required during the boot process to ensure the authenticity and
+ integrity of the device **before** providing it with potentially sensitive
+ artifacts such as certificates, key and device configuration.
+
+### Reference Documentation
+
+* External
+  * RFC: [RFC8572](https://datatracker.ietf.org/doc/html/rfc8572)
+  * Attestation: [Charra Draft](https://datatracker.ietf.org/doc/pdf/draft-ietf-rats-yang-tpm-charra-21#page=5)
+
+### Bootz Operation
+
+Devices are expected to perform a standard DHCP boot. The DHCP server passes
+ a boot option to the device for an endpoint (URL) from which the boot
+ package can be retrieved. The package returned by the endpoint consists of a
+ binary encoded protocol buffer containing all data for being able to
+ complete the boot process. In this context, “complete the boot process”
+ implies the device reaching a fully manageable state - with the relevant
+ gRPC services running, security operations(enrollment, attestation,
+ production credentials installed) are completed.
+
+Upon receiving the bootz protocol buffer, the device is responsible for
+unmarshalling the bootz message and distributing to the relevant system
+components for these artifacts to be acted upon. For example, configuration
+to a configuration manager component, keys and credentials to the relevant
+security components, etc.
+
+ The bootz payload will be encrypted via the TLS session underlying the gRPC service.
+
+After loading all the provided data on first boot the device is still not in
+ a trusted state, however it should have enough g\* services initialized to a
+ state where the device can be interrogated from a trusted system to enroll
+ the TPM and validate specific TPM values to attest the device.  Once
+ attested, the systems can install production configuration and certificates
+ into the device.
+
+## Detailed Design
+
+### Boot Procedure
+
+#### **API flow**
+
+1. DHCP Discovery of Bootstrap Server
+    1. Device sends DHCP messages, containing the mac-address of the active control card. The DHCP server has been configured with all possible mac-addresses of the device, and responds with the static IP address of the bootstrap server.
+    2. DHCP server also assigns an IP address and a gateway to the device.
+    3. The DHCP response option code should be 70 (Picked from the set of reserved/private use option codes).
```

Comment:
Sztp work flow uses option 136/143 in dhcp response message to inform redirect server information to client. Any reason why not use these?

Reply:
what is 136? i see 143 that looks right

Reply:
From the RFC text https://datatracker.ietf.org/doc/rfc8572/ 

8.2.  DHCPv6 SZTP Redirect Option

   The DHCPv6 SZTP Redirect Option is used to provision the client with
   one or more URIs for bootstrap servers that can be contacted to
   attempt further configuration.

      0                   1                   2                   3
      0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
     |       option-code (136)       |          option-length        |
     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
     .           bootstrap-server-list (variable length)             .
     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+

     * option-code: OPTION_V6_SZTP_REDIRECT (136)